src/components/Form.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import cn from 'classnames';
+
+import {
+  FormEvent,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { TodoContext, USER_ID } from '../providers/TodoProvider';
+import { TodoError } from '../types/TodoError';
+import { addTodo, editTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { FormContext } from '../providers/FormProvider';
+
+export const Form = () => {
+  const {
+    todos,
+    setTodos,
+    setError,
+    inputRef,
+  } = useContext(TodoContext);
+
+  const {
+    setPreparingTodoLabel,
+    setIsCreating,
+    setIsUpdating,
+    setIsToggleAll,
+    isToggleAll,
+  } = useContext(FormContext);
+
+  const [todoLabel, setTodoLabel] = useState('');
+  const [isBlockedInput, setIsBlockedInput] = useState(false);
+
+  const isActiveButton = todos.every(task => task.completed);
+
+  useEffect(() => {
+    setIsToggleAll(isActiveButton);
+  }, [isActiveButton, setIsToggleAll]);
+
+  useEffect(() => {
+    if (isBlockedInput === false) {
```

Comment:
```suggestion
    if (!isBlockedInput) {
```

can be a bit simplify